src/applications/travel-pay/components/complex-claims/pages/AgreementPage.jsx

```diff
@@ -1,24 +1,40 @@
 import React, { useState } from 'react';
 
 import { useNavigate, useParams } from 'react-router-dom-v5-compat';
-import {
-  VaCheckbox,
-  VaButtonPair,
-} from '@department-of-veterans-affairs/component-library/dist/react-bindings';
+import { useSelector, useDispatch } from 'react-redux';
+import { VaCheckbox } from '@department-of-veterans-affairs/component-library/dist/react-bindings';
 import TravelAgreementContent from '../../TravelAgreementContent';
+import TravelPayButtonPair from '../../shared/TravelPayButtonPair';
+import { submitComplexClaim } from '../../../redux/actions';
+import {
+  selectComplexClaim,
+  selectComplexClaimSubmissionState,
+} from '../../../redux/selectors';
 
 const AgreementPage = () => {
   const navigate = useNavigate();
+  const dispatch = useDispatch();
   const { apptId, claimId } = useParams();
+  const { data: claimData } = useSelector(selectComplexClaim);
+  const { isSubmitting } = useSelector(selectComplexClaimSubmissionState);
   const [isAgreementChecked, setIsAgreementChecked] = useState(false);
   const [isAgreementError, setIsAgreementError] = useState(false);
-  const onSubmit = () => {
+
+  const onSubmit = async () => {
     if (!isAgreementChecked) {
       setIsAgreementError(true);
     } else {
       setIsAgreementError(false);
-      // TODO Add logic for Submitting the claim
-      navigate(`/file-new-claim/${apptId}/${claimId}/confirmation`);
+      try {
```

Comment:
Nitpicky: The negatives here twist my brain a little. Could we do something like (either now or in the future):

```javascript
setIsAgreementError(!isAgreementChecked);

if (isAgreementChecked) {
  navigate(...);
  try {
    // ...
  catch (error) {
    // ...
  }
}
```